examples/example-gateway/clients/baz/baz.go

```diff
@@ -0,0 +1,237 @@
+// Code generated by zanzibar
+// @generated
+// TODO: (lu) generate
+
+package bazClient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/uber-go/zap"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/zanzibar/runtime"
+
+	zt "github.com/uber/zanzibar/runtime/tchannel"
+	"go.uber.org/thriftrw/wire"
+
+	"github.com/uber/zanzibar/examples/example-gateway/build/gen-code/github.com/uber/zanzibar/clients/baz/baz"
+)
+
+// TChanBaz is the interface that defines the server handler and client interface.
+type TChanBaz interface {
+	Call(ctx context.Context, reqHeaders map[string]string, r *baz.SimpleService_Call_Args) (map[string]string, *baz.BazResponse, error)
+	Simple(ctx context.Context, reqHeaders map[string]string) (map[string]string, error)
+	SimpleFuture(ctx context.Context, reqHeaders map[string]string) (map[string]string, error)
+}
+
+// BazClient is the client to to talk to its backend service.
+type BazClient struct {
+	SimpleServiceClient
+}
+
+// NewClient returns a new http client for service Bar.
+func NewClient(config *zanzibar.StaticConfig, gateway *zanzibar.Gateway) *BazClient {
+	// TODO: (lu) get clientName from config
+	// clientName := config.MustGetString("clients.baz.clientName")
+	clientName := "BazClient"
+
+	// TODO: (lu) client channel opts
+	ch, err := tchannel.NewChannel(clientName, nil)
```

Comment:
We do not want to allocate new channels here.

There needs to be a `gateway.Channel` and we should allocate a sub channel here.